src/plonkish/compiler/cell_manager.rs

```diff
@@ -231,24 +231,26 @@ impl CellManager for SingleRowCellManager {
 
 #[derive(Debug, Default, Clone)]
 pub struct MaxWidthCellManager {
-    max_width: usize,
+    max_width_advice: usize,
+    max_width_fixed: usize,
     same_height: bool,
 }
 
 impl MaxWidthCellManager {
-    pub fn new(max_width: usize, same_height: bool) -> Self {
+    pub fn new(max_width_advice: usize, max_width_fixed: usize, same_height: bool) -> Self {
         Self {
-            max_width,
+            max_width_advice,
+            max_width_fixed,
             same_height,
         }
     }
 }
 
 impl CellManager for MaxWidthCellManager {
     fn place<F>(&self, unit: &mut CompilationUnit<F>) {
-        if (!unit.shared_signals.is_empty() || !unit.fixed_signals.is_empty()) && !self.same_height
+        if (!unit.shared_signals.is_empty()) && !self.same_height
```

Comment:
@atsupontio this must be kept. Fixed signals should not work when same_height is disabled